### PR TITLE
feat(rmo): add osde2e tests [OSD-5524]

### PR DESCRIPTION
**Pull request by georgettica**

### CODE REVIEW

**Comment by georgettica** [comment]
shouldn't pass the tests, didn't solve them locally yet with `make check`

**Review by whereswaldon — CHANGES_REQUESTED** [review]
There are a few typos in some function names, but the bodies of those functions look reasonable. I didn't review the test code in depth, mostly because this PR is too large for that to be feasible. Thanks for moving those helpers so that other testers can use them!

**Comment by georgettica** [comment]
did all changes requested, passing the question "is this e2e test good enough for RMO" to the team for discussion

thanks @whereswaldon for your assisntance!

**Comment by georgettica** [comment]
@NautiluX, @RiRa12621  if you want to TAL at these..

I wasn't sure if:
- I should teardown everything and start it up for each test
- this one big test is good enough

aside of that I can phase it in:
1. only the base changes
2. add the custom e2e integration that is inspired from `make test-integration`

**Comment by whereswaldon** [comment]
Since you're consulting your team, I'm going to hold this:

/hold

That being said,

/lgtm

File: pkg/e2e/operators/routemonitor.go
Comment (NautiluX) [inline pkg/e2e/operators/routemonitor.go:83]: is this related to UWM?

File: pkg/e2e/operators/routemonitor.go
Comment (NautiluX) [inline pkg/e2e/operators/routemonitor.go:89]: guess you want to create a service for RMO here. Makes sense, but the message is confusing.

File: pkg/e2e/operators/routemonitor.go
Comment (NautiluX) [inline pkg/e2e/operators/routemonitor.go:81]: is this image managed?

File: pkg/e2e/operators/routemonitor.go
Comment (NautiluX) [inline pkg/e2e/operators/routemonitor.go:77]: I think this it does more than just prerequisite checking, right? Seems to do a full RMO test.

File: pkg/e2e/operators/routemonitor.go
Comment (NautiluX) [inline pkg/e2e/operators/routemonitor.go:118]:
```suggestion
			Expect(err).NotTo(HaveOccurred(), "Couldn't delete application route monitor")
```
```suggestion
			Expect(err).NotTo(HaveOccurred(), "Coudn't delete application route monitor")
```

**Review by NautiluX — CHANGES_REQUESTED** [review]
Like the readability of the test, mainly some of the messages are misleading. When those tests fail, the messages should make it easy to understand what's failing so the failure can be routed to us.

File: pkg/e2e/operators/routemonitor.go
Comment (georgettica) [reply pkg/e2e/operators/routemonitor.go:83]: It's related to verifying metrics  and availability of a route..
I think this can also be used as a application which can receive route access for synthetic probes

File: pkg/e2e/operators/routemonitor.go
Comment (georgettica) [reply pkg/e2e/operators/routemonitor.go:89]: Yeah, I'll fix it :(

File: pkg/e2e/operators/routemonitor.go
Comment (georgettica) [reply pkg/e2e/operators/routemonitor.go:81]: I wouldn't know, but I guessed it's a good step one?

File: pkg/e2e/operators/routemonitor.go
Comment (georgettica) [reply pkg/e2e/operators/routemonitor.go:77]: Yeah, copy pasta I will fix.
Thanks!

File: pkg/e2e/operators/routemonitor.go
Comment (georgettica) [reply pkg/e2e/operators/routemonitor.go]: found a a better step 1 nginx managed image

**Comment by georgettica** [comment]
@whereswaldon can you approve the PR?
I think the changes you requested were addressed and if not, feel free to update me

**Comment by NautiluX** [comment]
you're not planning to merge this before the MCC change is in right? Or should we remove/disable the test for the routemonitor to exist?

**Comment by georgettica** [comment]
@whereswaldon it seems the PR complains you still request changes, which is not the case
@NautiluX can this PR be merged as this is available on stage?

**Review by whereswaldon — APPROVED** [review]

**Comment by whereswaldon** [comment]
Looks like there are some unused variables in here. That's why CI is failing. https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_osde2e/747/pull-ci-openshift-osde2e-main-verify/1380175834341445632#1:build-log.txt%3A230

**Comment by georgettica** [comment]
Yeah, I know.. just haven't been on the computer..

Maybe also I could have a sit down with one of y'all to see how to test this on a cluster (so far I didn't get much luck)

**Comment by georgettica** [comment]
found this and will incorporate to my code.. still not sure why I need to do that but that's to another day

https://github.com/openshift/osde2e/blob/08efa2dfdcb1ddcd1d2db19225de28445b2f9978/pkg/e2e/operators/mustgather.go#L123-L126

**Comment by georgettica** [comment]
it worked and most of the issues were solved.

I have one last Q before merging..

for some reason [this](https://github.com/openshift/osde2e/pull/747/files#diff-)b9e52ef87dda6f8b11b1231478fd5f4d5529579b34b7ca9f30c33ab43eccce8bR139-R142) was required to make all of the tests pass.. is there a reason for it?

**Comment by georgettica** [comment]
nope, I add the routemonitor [here](https://github.com/openshift/osde2e/pull/747/files#diff-b9e52ef87dda6f8b11b1231478fd5f4d5529579b34b7ca9f30c33ab43eccce8bR109) and after I check if the resources were cleaned up

**Comment by georgettica** [comment]
just in case 

/test verify

File: pkg/e2e/operators/routemonitor.go
Comment (NautiluX) [inline pkg/e2e/operators/routemonitor.go:60]: What are those functions doing? Can we probably use a more speaking name?

**Review by NautiluX — CHANGES_REQUESTED** [review]
So we're now not relying on the RouteMonitor that's deployed via managed-cluster-config anymore right? Like it. I'm happy with the change overall, other than the two functions which look like they kept their WIP names.

**Comment by georgettica** [comment]
@NautiluX @whereswaldon 
I will merge the commits and it can be merged (added a TODO if we get to solve the issue sometime https://github.com/openshift/osde2e/pull/747/files#diff-b9e52ef87dda6f8b11b1231478fd5f4d5529579b34b7ca9f30c33ab43eccce8bR140)

**Review by NautiluX — COMMENTED** [review]
looks good to me now. Were you able to verify those tests work? If so I think we're good to merge after rebasing!

**Comment by georgettica** [comment]
I was @NautiluX , but I had one weird k8s issue there I wasn't sure if to patch around or fix

**Comment by georgettica** [comment]
@NautiluX it's ready!

**Comment by georgettica** [comment]
/unhold
it works now :)

**Pull request merged**
